ucp/_libs/arr.pyx

```diff
@@ -0,0 +1,287 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+# cython: language_level=3
+
+
+from cpython.buffer cimport PyBuffer_IsContiguous
+from cpython.mem cimport PyMem_Free, PyMem_Malloc
+from cpython.memoryview cimport (
+    PyMemoryView_FromObject,
+    PyMemoryView_GET_BUFFER,
+)
+from cpython.ref cimport Py_INCREF
+from cpython.tuple cimport PyTuple_New, PyTuple_SET_ITEM
+from cython cimport auto_pickle, boundscheck, wraparound
+from libc.stdint cimport uintptr_t
+from libc.string cimport memcpy
+
+try:
+    from numpy import dtype as numpy_dtype
+except ImportError:
+    numpy_dtype = None
+
+
+cdef dict itemsize_mapping = {
+    intern("|b1"): 1,
+    intern("|i1"): 1,
+    intern("|u1"): 1,
+    intern("<i2"): 2,
+    intern(">i2"): 2,
+    intern("<u2"): 2,
+    intern(">u2"): 2,
+    intern("<i4"): 4,
+    intern(">i4"): 4,
+    intern("<u4"): 4,
+    intern(">u4"): 4,
+    intern("<i8"): 8,
+    intern(">i8"): 8,
+    intern("<u8"): 8,
+    intern(">u8"): 8,
+    intern("<f2"): 2,
+    intern(">f2"): 2,
+    intern("<f4"): 4,
+    intern(">f4"): 4,
+    intern("<f8"): 8,
+    intern(">f8"): 8,
+    intern("<f16"): 16,
+    intern(">f16"): 16,
+    intern("<c8"): 8,
+    intern(">c8"): 8,
+    intern("<c16"): 16,
+    intern(">c16"): 16,
+    intern("<c32"): 32,
+    intern(">c32"): 32,
+}
+
+
+@auto_pickle(False)
+cdef class Array:
```

Comment:
Would be good with some doc string here

Reply:
I've added something simple as a starting point. Though if there's more things you would like to see here, happy to fill this out further 🙂